### ISSUE TITLE
Cop Config reads "on" as a boolean (true)

### ISSUE THREAD

**Issue by corroded** [rubocop/rubocop#9596]
The Rubocop config loader somehow interprets the string 'on' as the boolean `true`.

--------

## Expected behavior

I expect adding a config with the string 'on' will make the cop read it as 'on'

## Actual behavior

When adding 'on' as an option, the config reads it as 'true'.

## Steps to reproduce the problem

I have investigated this but I think it's somewhere in the YAML loader or the `Config` / `ConfigLoader` classes. To reproduce, I actually used `rubocop-rspec`'s `RSpec/ContextWording` cop.

The `ContextWording` cop has a configurable option called `Prefixes`:

```
RSpec/ContextWording:
  Enabled: true
  Prefixes:
    - when
    - with
    - without
    - on
```

This is loaded by Rubocop Config as:

```
   "Prefixes"=>["when", "with", "without", true],
```

### For more detailed debugging:

I added a `pry` in the initializer of config.rb

~/.rbenv/versions/2.6.6/lib/ruby/gems/2.6.0/gems/rubocop-1.11.0/lib/rubocop/config.rb
```
    def initialize(hash = {}, loaded_path = nil)
      @loaded_path = loaded_path
      @for_cop = Hash.new do |h, cop|
        qualified_cop_name = Cop::Registry.qualified_cop_name(cop, loaded_path)
        cop_options = self[qualified_cop_name] || {}
        cop_options['Enabled'] = enable_cop?(qualified_cop_name, cop_options)
        h[cop] = cop_options
      end
      binding.pry
      @hash = hash
      @validator = ConfigValidator.new(self)
    end
```
This is part of the output of `hash`:

![image](https://user-images.githubusercontent.com/46548/110957646-c3470580-8386-11eb-938c-720ed00d4126.png)

I also updated `config/default.yml` to have:

```
RSpec/ContextWording:
  Description: Checks that `context` docstring starts with an allowed prefix.
  Enabled: true
  Prefixes:
    - when
    - with
    - without
    - on
  VersionAdded: '1.20'
  VersionChanged: 1.20.1
  StyleGuide: https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ContextWording
```

## RuboCop version
```
- rubocop 1.11.0 # spotted as early as 0.72.0 -- this is our version, but I upgraded and it still happens
- rubocop-rspec 2.2.0 # spotted as early as 1.36.0 -- our version
```

**Comment by koic** [comment]
This is an expected behavior based on the YAML specification.

> A Boolean represents a true/false value. Booleans are formatted as English words (“true”/“false”, “yes”/“no” or “on”/“off”) for readability and may be abbreviated as a single character “y”/“n” or “Y”/“N”.

https://yaml.org/type/bool.html

JFYI, `yes` is similar to `on`. These should be enclosed in quotes in order to be treated as strings.

```ruby
% cat example.rb
require 'yaml'

puts YAML.load('key: "on"')
puts YAML.load('key: on')
puts YAML.load('key: "yes"')
puts YAML.load('key: yes')

% ruby example.rb
{"key"=>"on"}
{"key"=>true}
{"key"=>"yes"}
{"key"=>true}
```

Please try the following:

```diff
 RSpec/ContextWording:
   Enabled: true
   Prefixes:
     - when
     - with
     - without
-    - on
+    - 'on'
```

**Comment by corroded** [comment]
Thanks for this! I figured it might be `yaml` - should've tested for that as well. Thank you 💯
